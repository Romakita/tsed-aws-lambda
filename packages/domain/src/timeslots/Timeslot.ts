import { Description, Example, Format, Groups, JsonFormatTypes, MaxLength, MinLength, Name, Required, RequiredGroups } from "@tsed/schema";

import { AllowEmpty } from "../decorators/AllowEmpty.js";

export class Timeslot {
  @Format(JsonFormatTypes.UUID)
  @Required()
  @RequiredGroups("update")
  id: string;

  @Required()
  @Name("start_date")
  @Format("date-time")
  startDate: Date;

  @Required()
  @Name("end_date")
  @Format("date-time")
  endDate: Date;

  @Required()
  @Example("Hiking")
  @MinLength(3)
  @MaxLength(100)
  label: string;

  @Required()
  @AllowEmpty()
  @Description("Activity or Dresscode in HTML format")
  @MinLength(3)
  @MaxLength(500)
  description: string = "";

  @Required()
  @Name("created_at")
  @Format("date-time")
  @Groups("!update", "!create")
  createdAt: Date = new Date();

  @Required()
  @Name("updated_at")
  @Format("date-time")
  @Groups("!update", "!create")
  updatedAt: Date = new Date();
}
