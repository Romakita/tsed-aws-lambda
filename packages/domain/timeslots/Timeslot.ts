import { Description, Example, Format, MaxLength, MinLength, Name, Required } from "@tsed/schema"

import { AllowEmpty } from "../decorators/AllowEmpty.js"
import { BaseDocument } from "../document/BaseDocument.js"

export class Timeslot extends BaseDocument {
  @Required()
  @Name("start_date")
  @Format("date-time")
  startDate: Date

  @Required()
  @Name("end_date")
  @Format("date-time")
  endDate: Date

  @Required()
  @Example("Hiking")
  @MinLength(3)
  @MaxLength(100)
  label: string

  @Required()
  @AllowEmpty()
  @Description("Activity or Dresscode in HTML format")
  @MinLength(3)
  @MaxLength(500)
  description: string = ""
}
