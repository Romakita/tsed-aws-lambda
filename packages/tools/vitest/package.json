{
  "name": "@project/vitest",
  "version": "1.0.0",
  "description": "Vitest utilities for project",
  "main": "src/index.ts",
  "typings": "src/index.ts",
  "devDependencies": {
    "@aws-sdk/client-lambda": "^3.540.0"
  }
}
