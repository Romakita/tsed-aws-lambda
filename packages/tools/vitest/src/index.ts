export * from "./utils/createRequest.js"
export * from "./utils/getAuthTokenFixture.js"
export * from "./utils/invokeLambda.js"
