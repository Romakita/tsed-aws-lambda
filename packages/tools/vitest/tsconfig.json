{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "composite": true,
    "noEmit": false,
    "allowJs": false,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "strict": true,
    "types": ["vite/client", "vitest"]
  },
  "files": [],
  "include": [],
  "references": [
    {
      "path": "./tsconfig.node.json"
    }
  ]
}
