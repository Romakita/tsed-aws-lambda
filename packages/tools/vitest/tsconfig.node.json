{
  "extends": "../../../tsconfig.base.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./dist",
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "nodeNext",
    "types": ["node", "vitest", "vitest/globals", "vitest/importMeta"]
  },
  "include": ["src/**/*.ts"],
  "exclude": ["dist", "src/**/*.spec.ts", ".esbuild"]
}
