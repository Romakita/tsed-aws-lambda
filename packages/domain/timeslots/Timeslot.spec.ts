import { deserialize } from "@tsed/json-mapper"
import { getJsonSchema } from "@tsed/schema"
import { afterEach, beforeEach } from "vitest"

import { Timeslot } from "./Timeslot.js"

describe("Timeslot", () => {
  beforeEach(() => {
    vi.useFakeTimers({ now: new Date("2024-04-02T07:03:28.388Z") })
  })
  afterEach(() => {
    vi.useRealTimers()
  })
  describe("json schema", () => {
    it("should return the expected json schema", () => {
      expect(getJsonSchema(Timeslot)).toMatchSnapshot()
    })
  })

  describe("deserialize()", () => {
    it("should deserialize the given data", () => {
      const result = deserialize<Timeslot>(
        {
          id: "id",
          label: "Hiking",
          start_date: "2021-01-01T10:00:00.000Z",
          end_date: "2021-01-01T11:00:00.000Z"
        },
        {
          type: Timeslot
        }
      )

      expect(result).toBeInstanceOf(Timeslot)
      expect(result).toEqual({
        description: "",
        id: "id",
        label: "Hiking",
        startDate: new Date("2021-01-01T10:00:00.000Z"),
        endDate: new Date("2021-01-01T11:00:00.000Z"),
        createdAt: new Date("2024-04-02T07:03:28.388Z"),
        updatedAt: new Date("2024-04-02T07:03:28.388Z")
      })
    })
  })
})
