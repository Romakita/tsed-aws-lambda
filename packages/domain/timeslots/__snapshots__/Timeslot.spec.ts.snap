// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Timeslot > json schema > should return the expected json schema 1`] = `
{
  "definitions": {
    "Date": {
      "type": "object",
    },
  },
  "properties": {
    "created_at": {
      "$ref": "#/definitions/Date",
      "minLength": 1,
    },
    "description": {
      "default": "",
      "description": "Activity or Dresscode in HTML format",
      "maxLength": 500,
      "minLength": 3,
      "type": "string",
    },
    "end_date": {
      "$ref": "#/definitions/Date",
      "minLength": 1,
    },
    "id": {
      "format": "uuid",
      "type": "string",
    },
    "label": {
      "examples": [
        "Hiking",
      ],
      "maxLength": 100,
      "minLength": 3,
      "type": "string",
    },
    "start_date": {
      "$ref": "#/definitions/Date",
      "minLength": 1,
    },
    "updated_at": {
      "$ref": "#/definitions/Date",
      "minLength": 1,
    },
  },
  "required": [
    "created_at",
    "updated_at",
    "start_date",
    "end_date",
    "label",
    "description",
  ],
  "type": "object",
}
`;
